php/libraries/TimePointData.php

```diff
@@ -0,0 +1,106 @@
+<?php declare(strict_types=1);
+/**
+ * A TimePointData object is a DTO (data transfer object) to
+ * store a representation of data associated with a TimePoint
+ * object in LORIS.
+ *
+ * The TimePointData may be partially loaded, in which case any
+ * getter trying to access an unknown will throw an exception.
+ * This allows modules to only load relevant data in a query,
+ * and avoid the overhead of the TimePoint::singleton() constructor.
+ *
+ * For instance, a query that selected CenterID and ProjectID on
+ * the database could instantiate a TimePoint as:
+ *    $visit = new TimePoint(new TimePointData(null, $projectid, $centerid));
+ *
+ * with data loaded from the query, and then call
+ * `$visit->isAccessibleBy($user)` without having to load all the
+ * over data that TimePoint::singleton() does. If an attempt is
+ * made to access a property not in the DTO (session in this example),
+ * an exception is thrown, rather than silently returning the incorrect
+ * value.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class TimePointData
```

Comment:
To me, in terms of nomenclature, this object makes more sense to me as the `TimePoint`, and the current "TimePoint" is more akin to a "TimePointController" or "TimePointHandler".

Reply:
I don't think we can change the name of the `TimePoint` class at this point, but it could eventually be split into further classes (it's just not my concern here because this is being driven by needing more efficiency for `isAccessibleBy` for data querying.)